Compilation error with the Sugon environment

### Describe the bug

Compilation error with the Sugon environment:
```
In file included from /work/home/acwuilbxmy/4_denghui/deepmd-kit/abacus-develop/source/module_base/libm/exp.cpp:6:
/opt/rh/devtoolset-7/root/usr/include/c++/7/math.h:65:12: error: target of using declaration conflicts with declaration already in scope
using std::isinf;
           ^
/opt/rh/devtoolset-7/root/usr/include/c++/7/cmath:599:3: note: target of using declaration
  isinf(double __x)
  ^
/usr/include/bits/mathcalls.h:202:19: note: conflicting declaration
__MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
                  ^
In file included from /work/home/acwuilbxmy/4_denghui/deepmd-kit/abacus-develop/source/module_base/libm/exp.cpp:6:
/opt/rh/devtoolset-7/root/usr/include/c++/7/math.h:66:12: error: target of using declaration conflicts with declaration already in scope
using std::isnan;
           ^
/opt/rh/devtoolset-7/root/usr/include/c++/7/cmath:626:3: note: target of using declaration
  isnan(double __x)
  ^
/usr/include/bits/mathcalls.h:235:19: note: conflicting declaration
__MATHDECL_1 (int,isnan,, (_Mdouble_ __value)) __attribute__ ((__const__));
                  ^
2 errors generated.
make[2]: *** [source/module_base/CMakeFiles/base.dir/build.make:748: source/module_base/CMakeFiles/base.dir/libm/exp.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:893: source/module_base/CMakeFiles/base.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
(deepmd-kit) [acwuilbxmy@login01:build]$ vi /work/home/acwuilbxmy/4_denghui/deepmd-kit/abacus-develop/source/module_base/libm/exp.cpp
(deepmd-kit) [acwuilbxmy@login01:build]$ 
(deepmd-kit) [acwuilbxmy@login01:build]$ 
(deepmd-kit) [acwuilbxmy@login01:build]$ make 
[  4%] Built target relax
[  4%] Built target driver
[  9%] Built target device_rocm
[ 13%] Built target device
Consolidate compiler generated dependencies of target base
[ 13%] Building CXX object source/module_base/CMakeFiles/base.dir/libm/exp.cpp.o
[ 13%] Building CXX object source/module_base/CMakeFiles/base.dir/libm/sincos.cpp.o
In file included from /work/home/acwuilbxmy/4_denghui/deepmd-kit/abacus-develop/source/module_base/libm/sincos.cpp:6:
/opt/rh/devtoolset-7/root/usr/include/c++/7/math.h:65:12: error: target of using declaration conflicts with declaration already in scope
using std::isinf;
           ^
/opt/rh/devtoolset-7/root/usr/include/c++/7/cmath:599:3: note: target of using declaration
  isinf(double __x)
  ^
/usr/include/bits/mathcalls.h:202:19: note: conflicting declaration
__MATHDECL_1 (int,isinf,, (_Mdouble_ __value)) __attribute__ ((__const__));
                  ^
In file included from /work/home/acwuilbxmy/4_denghui/deepmd-kit/abacus-develop/source/module_base/libm/sincos.cpp:6:
/opt/rh/devtoolset-7/root/usr/include/c++/7/math.h:66:12: error: target of using declaration conflicts with declaration already in scope
using std::isnan;
           ^
/opt/rh/devtoolset-7/root/usr/include/c++/7/cmath:626:3: note: target of using declaration
  isnan(double __x)
  ^
/usr/include/bits/mathcalls.h:235:19: note: conflicting declaration
__MATHDECL_1 (int,isnan,, (_Mdouble_ __value)) __attribute__ ((__const__));
                  ^
2 errors generated.
make[2]: *** [source/module_base/CMakeFiles/base.dir/build.make:762: source/module_base/CMakeFiles/base.dir/libm/sincos.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:893: source/module_base/CMakeFiles/base.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Expected behavior

_No response_

### To Reproduce

Compilation commands:
```
CC=clang CXX=clang++ cmake -DUSE_ROCM=ON -DENABLE_LCAO=OFF -DUSE_OPENMP=OFF -DFFTW3_DIR=/work/home/acwuilbxmy/4_denghui/deepmd-kit/fftw-3.3.10 -DLAPACK_DIR=/work/home/acwuilbxmy/4_denghui/deepmd-kit/OpenBLAS/lib .. 
```

### Environment

Here's the loaded modules : 
```
Currently Loaded Modulefiles:
  1) compiler/devtoolset/7.3.1   3) compiler/cmake/3.23.1       5) apps/anaconda3/5.2.0
  2) mpi/hpcx/gcc-7.3.1          4) compiler/dtk/22.10
```

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [X] Verify the issue is not a duplicate.
- [X] Describe the bug.
- [X] Steps to reproduce.
- [X] Expected behavior.
- [X] Error message.
- [X] Environment details.
- [X] Additional context.
- [X] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ option(ENABLE_LIBCOMM "Enable communicate with LibComm." OFF)
 option(ENABLE_PAW "Enable PAW calculation" OFF)
 option(ENABLE_MPI "Enable compilation with or without MPI." ON)
 option(USE_ELPA "Enable ELPA" ON)
-option(USE_ABACUS_LIBM "Build libmath from source to speed up." ON)
+option(USE_ABACUS_LIBM "Build libmath from source to speed up." OFF)
 option(GIT_SUBMODULE "Check submodules during build" ON)
 option(DEBUG_INFO "Print message for developers to debug." OFF)
 # Do not enable it if generated code will run on different CPUs

